spinalcordtoolbox/scripts/sct_check_dependencies.py

```diff
@@ -347,6 +347,20 @@ def main(argv=None):
             print_fail()
             print(err)
 
+    # Check version of FSLeyes
+    print_line('Check FSLeyes version')
+    cmd = 'fsleyes --version'
+    status, output = run_proc(cmd, verbose=0, raise_exception=False)
+    # Fetch only version number (full output of 'fsleyes --version' is 'fsleyes/FSLeyes version 0.34.2')
+    fsleyes_version = output.split()[2]
+    if status == 0:
+        print_ok(more=(" (%s)" % fsleyes_version))
+    else:
+        print_fail(f" ({output})")
```

Comment:
My only concern is that outputting a "FAIL" seems a little strong, considering that FSLeyes isn't required to use SCT. Even "WARNING" is a little strong... I worry that a user may see this and think that the installation is broken.

How do you feel about skipping the check entirely if FSLeyes isn't installed?

Reply:
On the other hand, it might help us (and possibly help users) to know if FSLeyes is installed or not, in case they are running into FSLeyes-related issues. I agree "FAIL/WARNING" is strong though. Maybe we can do something like this?
```python
    if status == 0:
        print_ok(more=(" (%s)" % fsleyes_version))
    else:
        print(f" Not installed. While FSLeyes is not a requirement of SCT, we recommend to install it to easily visualize processing outputs and/or to use SCT within FSLeyes. More info at: https://spinalcordtoolbox.com/en/latest/user_section/fsleyes.html")
```

Reply:
I like that! 

Though, I'm curious how that long message would look given the "`..... [ ]`" formatting from `sct_check_dependencies`.

Reply:
> Though, I'm curious how that long message would look given the "..... [ ]" formatting from sct_check_dependencies.

Yes, I thought about that... we would need to check how the output looks like 😊

Reply:
Thanks! I agree that "FAIL" was too strong since FSL is not required for SCT installation. I switched from "FAIL" to Julien's message in my last commit.
There could be also situation, that user can have FSLeyes installed but can run into some another error (like the user [here](https://forum.spinalcordmri.org/t/problem-with-fsleyes/715/7) due to `sct_plugin.py`). This potential problem should be caught by `sct_check_dependencies -complete` I hope.

Reply:
:smile: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3428/checks?check_run_id=2875654109#step:4:59

Reply:
> There could be also situation, that user can have FSLeyes installed but can run into some another error (like the user [here](https://forum.spinalcordmri.org/t/problem-with-fsleyes/715/7) due to `sct_plugin.py`). This potential error should be caught by `sct_check_dependencies -complete` I hope.

This could be addressed by two-level check like this:
```python
# Check version of FSLeyes
print_line('Check FSLeyes version')
cmd = 'which fsleyes'
status, output = run_proc(cmd, verbose=0, raise_exception=False)
# Check if FSLeyes is installed
if status == 0:
    cmd = 'fsleyes --version'
    status, output = run_proc(cmd, verbose=0, raise_exception=False)
    if status == 0:
        # Fetch only version number (full output of 'fsleyes --version' is 'fsleyes/FSLeyes version 0.34.2')
        fsleyes_version = output.split()[2]
        print_ok(more=(" (%s)" % fsleyes_version))
    else:
        print("FSLeyes looks to be installed but some error occurred. Run sct_check_dependencies -complete to see "
              "the error.")
else:
    print("Not installed. While FSLeyes is not a requirement of SCT, we recommend to install it to easily "
          "visualize processing outputs and/or to use SCT within FSLeyes. More info at: "
          "https://spinalcordtoolbox.com/en/latest/user_section/fsleyes.html")
if complete_test:
    print('>> ' + cmd)
    print((status, output), '\n')
```

Reply:
> Yes, I thought about that... we would need to check how the output looks like 😊

The output is wrapped based on terminal's windows width:

<img width="1176" alt="image" src="https://user-images.githubusercontent.com/39456460/122781599-447c3580-d2b0-11eb-845a-31c54624b7ac.png">

Reply:
Good thinking with the two-level check. :slightly_smiling_face: 

I'd recommend using [`shutil.which`](https://stackoverflow.com/a/34177358) instead of `run_proc(which)` for [portability reasons](https://stackoverflow.com/a/37056347), though.

Reply:
@joshuacwnewton Thank you for recommendations! I implemented the second one in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3428/commits/259e06fa636c25381404d408a588e3c1f8d103ea. I am now catching for `RuntimeError` exception (FSLeyes is not installed), `AttributeError` exception ([sct_plugin.py related exception](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3426#issue-925060441)) and all other exceptions using `except Exception`.